src/main/java/org/schemaspy/service/TableService.java

```diff
@@ -353,14 +353,13 @@ protected void addForeignKey(Database db, Table table, String fkName, String fkC
             Table parentTable = tables.get(pkTableName);
 
             String parentContainer = pkSchema != null ? pkSchema : pkCatalog != null ? pkCatalog : db.getName();
-            String catalog = commandLineArguments.getCatalog();
-            String baseContainer = config.getSchema() != null ? config.getSchema() : catalog != null ? catalog : db.getName();
+            String childContainer = table.getSchema() != null ? table.getSchema() : table.getCatalog() != null ? table.getCatalog() : db.getName();
```

Comment:
Doesn't Table have getContainer?
Might be good to extract method so that the way it's creates can be identical in logic.

Reply:
Yeah, it's strange.
Table has 
`this.container = schema != null ? schema : catalog != null ? catalog : db.getName()`
RemoteTable has
`this.baseContainer = baseContainer;`
Since RemoteTable is a subclass of Table it also has container from Table.

And baseContainer is set with all kinds of values that are either for the remoteTable or for the childTable.

Also baseContainer can be either Schema or Catalog.

Issues to create:
1, Rename baseContainer in RemoteTable to something in the line of childContainer, same goes for parameters used to set the field.
2, Refactor to make it catalog aware (Catalog|Database).Schema